File: 16-3sum-closest/16-3sum-closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) 
    {
        int n=nums.size();
        int ans=0;
        int diff=INT_MAX;
        sort(nums.begin(),nums.end());
        for(int i=0;i<n-1;i++)
        {
            int l=i+1;
            int r=n-1;
            while(l<r)
            {
                int sum=nums[i]+nums[l]+nums[r];
                if(abs(target-sum)<diff)
                {
                    diff=abs(target-sum);
                    ans=sum;
                }
                if(sum>target)
                {
                    r--;
                }
                else
                {
                    l++;
                }
            }
        }
        return ans;
    }
};

File: 16-3sum-closest/16-3sum-closest.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {
    nums.sort((a,b)=>a-b);
    let n=nums.length;
    let result=10000000;
    let diff=100000000;
    for(let i=0;i<n-2;i++){
        let l=i+1;
        let r=n-1;
        while(l<r){
            let sum=nums[i]+nums[l]+nums[r];
            if(Math.abs(sum-target)<Math.abs(result-target)){
                result=sum;
            }
            if(sum<target){
                l++;
            }
            else{
                r--;
            }
        }
    }
    return result;
};